src/olympia/search/filters.py

```diff
@@ -266,11 +274,14 @@ def filter_queryset(self, request, qs, view):
 
         for filter_class in self.available_filters:
             try:
-                filter_ = filter_class(request)
-                if filter_.query_param in request.GET:
+                # Initialize the filter class if its query parameter is present
+                # in the request, otherwise don't, to avoid  raising exceptions
+                # because of missing params in complex filters.
+                if filter_class.query_param in request.GET:
+                    filter_ = filter_class(request)
                     must.extend(filter_.get_es_filter())
-            except ValueError:
-                continue
+            except ValueError as exc:
+                raise serializers.ValidationError(*exc.args)
```

Comment:
I only saw tests testing that `ValidationError` is raised correctly but not verifying that `*exc.args` and thus the correct message is correctly forwarded.